File: concurrency/c/threads/calculate_number_in_parallel.c
#include <stdio.h>
#include <stdlib.h>
#include <pthread.h>

void* square_of_number(void* arg) {
    int number = *(int*)arg;
    int result = number*number;
    printf("Square of %d is %d!\n", number, result);
    
    return NULL;
}

int main() {
    pthread_t threads[3];
    int threads_ids[3];

    for (int i=0; i<10; i++){
        threads_ids[i] = i+1;
        if (pthread_create(&threads[i], NULL, square_of_number, &threads_ids[i]) != 0) {
            perror("Failed to create thread");
            return 1;
        }
    }

    for (int i=0; i<10; i++){
        if (pthread_join(threads[i], NULL) != 0) {
            perror("Failed to join thread");
            return 1;
        }
    }

    return 0;
}

File: concurrency/python/threads/calculate_number_in_parallel.py
import threading

def square_of_number(number: int):
    result = number**2
    print(f"Square of {number} is {result}!")
    return result


def main():
    numbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9] 
    threads = []

    for num in numbers:
        thread = threading.Thread(target=square_of_number, args=(num, ))
        threads.append(thread)
        thread.start()

    for thread in threads:
        thread.join()


if __name__ == "__main__":
    main()
